src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -830,9 +830,25 @@ export class MatAutocompleteTrigger
     // the opposite end has rounded corners. We apply a CSS class to swap the
     // border-radius based on the overlay position.
     const panelClass = this._aboveClass;
+
+    /**
+     * Calculate the vertical offset for the overlay when the input field has an outline appearance.
+     * In the outline appearance, the overlay should have extra space to display the label correctly.
+     */
+    const outlineHeight =
```

Comment:
This logic will measure the floating label height even if the form field isn't using the `outlined` appearance. This should be guarded with the `hasOutline` check since it has performance implications.

Reply:
Thank you for pointing out the need to optimize this logic. I've first checked _hasOutline() to enhance performance when measuring the floating label height in cases where the form field isn't using the outlined appearance.